rotkehlchen/chain/evm/tokens.py

```diff
@@ -96,9 +96,16 @@ def generate_multicall_chunks(
 
 
 class EvmTokens(metaclass=ABCMeta):
-    def __init__(self, database: 'DBHandler', evm_inquirer: 'EvmNodeInquirer'):
+    def __init__(
+            self,
+            database: 'DBHandler',
+            evm_inquirer: 'EvmNodeInquirer',
+            # mapping of owner addresses to proxy addresses
+            proxies_mapping: Optional[dict[ChecksumEvmAddress, ChecksumEvmAddress]] = None,
+    ):
         self.db = database
         self.evm_inquirer = evm_inquirer
+        self.proxies_mapping = proxies_mapping if proxies_mapping is not None else {}
```

Comment:
Why do this?

Why not just use `self.evm_inquirer.proxies_inquirer.get_accounts_having_proxy() at the start of each function you need this? This way you can use the requery period.

```suggestion
```